src/main/java/com/icthh/xm/ms/balance/service/BalanceService.java

```diff
@@ -357,7 +358,8 @@ private BalanceChangeEvent createBalanceChangeEvent(Balance balance, OperationTy
         event.setBalanceKey(balance.getKey());
         event.setBalanceTypeKey(balance.getTypeKey());
         event.setBalanceEntityId(balance.getEntityId());
-        event.setExecutedByUserKey(authContextHolder.getContext().getRequiredUserKey());
+        event.setExecutedByUserKey(authContextHolder.getContext().getUserKey()
+            .orElseGet(()-> authContextHolder.getContext().getLogin().orElse(StringUtils.EMPTY)));
```

Comment:
May be it is much simplier to move ()-> authContextHolder.getContext().getLogin().orElse(StringUtils.EMPTY)) to private function

In this case
event.setExecutedByUserKey(authContextHolder.getContext().getUserKey()
            .orElseGet(this::getDefaultLogin);

....
protected getDefaultLogin() ....

Reply:
IMO, it'll be kind of redundant method in not really small class. And i do not really like  `getDefaultLogin()` which returns empty string in case of empty login (nobody would have guess about such method behavior ).
But i have shorted the line by extracting new variable. Check out my latest commit

Reply:
**Empty** setExecutedByUserKey is my other concern. It is better to see who invoked charging event (person, system, whatever...)

Reply:
JWT for systems has **client_id**: "PUBLIC_SITE" for users **user_key**: "recycler" and  **client_id**: "webapp"